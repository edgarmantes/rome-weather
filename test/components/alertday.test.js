import React from 'react'
import AlertDay from '../../src/components/alertday';
import renderer from 'react-test-renderer';

describe('<AlertDay />' , () => {

	it('Renders AlertDay component properly', () => {
		shallow(<AlertDay/>)
	});

	it("Renders with correct props.day value", () => {
		const day = "Friday";
		const wrapper = mount(<AlertDay day={day} />);
		expect(wrapper.find(".alert-day").text()).toEqual(day)
	})

	it("Renders with correct props.date value", () => {
		const day = "17";
		const wrapper = mount(<AlertDay date={day} />);
		expect(wrapper.find(".alert-date").text()).toEqual(day)
	})

	it("Renders with correct props.fahr value", () => {
		const fahr = "80";
		const output = "high of: 80";
		const wrapper = mount(<AlertDay fahr={fahr} />);
		expect(wrapper.find(".alert-temp").text()).toEqual(output)
	})

	it("Renders with correct props.date value", () => {
		const unit = "&#8457";
		const output = "high of: &#8457";
		const wrapper = mount(<AlertDay unit={unit} />);
		expect(wrapper.find(".alert-temp").text()).toEqual(output)
	})


	it('Snapshot of AlertDay', () => {
        const renderedValue =  renderer.create(<AlertDay />).toJSON()
        expect(renderedValue).toMatchSnapshot();
    });

});


