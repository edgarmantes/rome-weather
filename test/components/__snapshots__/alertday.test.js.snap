// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AlertDay /> Snapshot of AlertDay 1`] = `
<li
  className="alertday-item{key}"
>
  <p
    className="alert-day"
  >
    <span
      className="alert-date"
    />
  </p>
  <p
    className="alert-temp"
  >
    high of: 
  </p>
</li>
`;
